### ISSUE TITLE
What value should bin_score converge to ?

### ISSUE THREAD

**Issue by XuyangBai** [magicleap/SuperGluePretrainedNetwork#44]
Thanks for your great work. When I trained SuperGlue on my own dataset, I found the `bin_score` keep increasing at a relatively stable pace instead of converging to a dataset-specific value. I wonder is this a normal situation? If I understand correctly, the larger `bin_score` is, the more features will be considered as non-matching. So the extreme case is the `bin_score` is much larger than all the value in the `scores` matrix, which will lead to no matches surviving after `mscores0 > self.config['match_threshold']`. Could you provide some explanations?

Best,
Xuyang

**Comment by Skydes** [comment]
Your interpretation is correct. If you training data has way more unmatchable points than matches, increasing the bin score is a lazy solution to minimize the loss. One way to prevent this is to balance the positive and negative terms, [as explained here](https://github.com/magicleap/SuperGluePretrainedNetwork/issues/31#issuecomment-684986482).

**Comment by Skydes** [comment]
Hi @XuyangBai , could you please close this issue if your problem is solved? Thank you.

**Comment by XuyangBai** [comment]
Sure, thanks for your reply.
